src/WebJobs.Script/Content/Script/functions.js

```diff
@@ -13,6 +13,8 @@ module.exports = {
 function globalInitialization(context, callback) {
     process.on('uncaughtException', function (err) {
         context.handleUncaughtException(err.stack);
+        // delay exit to allow .net unhandled exception logging to complete
+        setTimeout(() => process.exit(1), 500);
```

Comment:
So after more digging you're right, this is a weird timing issue.  Debug a test, logs appear, run a test / function & they don't, so this 'fix' did nothing.